apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloCall.java

```diff
@@ -210,61 +211,69 @@ private RealApolloCall(Builder<T> builder) {
         .build();
   }
 
-  final class AsyncCall implements ApolloInterceptor.CallBack {
-
-    private final Callback<T> responseCallback;
-
-    private AsyncCall(Callback<T> callback) {
-      this.responseCallback = callback;
-    }
-
-    @Override public void onResponse(@Nonnull final ApolloInterceptor.InterceptorResponse response) {
-      try {
+  @Nonnull @Override public Operation operation() {
+    return operation;
+  }
 
-        if (responseCallback == null) {
-          return;
-        }
-        if (canceled) {
-          responseCallback.onCanceledError(new ApolloCanceledException("Canceled"));
-          return;
-        }
-        if (queryFetcher.isPresent()) {
-          queryFetcher.get().refetchAsync(new QueryFetcher.OnFetchCompleteCallback() {
-            @Override public void onFetchComplete() {
-              //noinspection unchecked
-              responseCallback.onResponse(response.parsedResponse.get());
-            }
-          });
-        } else { //noinspection unchecked
-          responseCallback.onResponse(response.parsedResponse.get());
+  private void refetchQueryWatchers() {
```

Comment:
Maybe this goes back to the discussion we had a while ago, I've gotten a little lost, but do we force user's to be watching a query in able to refetch it?

I've also envisioned them to be separate concepts -- you can watch whatever you want, and whenever a dependent key changes you get notified. And when you mutate, you can pass a set of real query objects to refetch -- these will have no direct listeners, _but_ if you are watching somewhere, the normal watcher flow will trigger and notify.